python/pyarrow/tests/test_io.py

```diff
@@ -105,6 +105,141 @@ def test_python_file_read():
     f.close()
 
 
+def test_python_file_readall():
+    data = b'some sample data'
+
+    buf = BytesIO(data)
+    with pa.PythonFile(buf, mode='r') as f:
+        assert f.readall() == data
+
+
+def test_python_file_readinto():
+    length = 10
+    data = b'some sample data longer than 10'
+    array = np.empty(length, dtype=np.uint8)
+    dst_buf = memoryview(array)
+    src_buf = BytesIO(data)
+
+    with pa.PythonFile(src_buf, mode='r') as f:
+        assert f.readinto(dst_buf) == 10
+
+        assert array[:length].tobytes() == data[:length]
+        assert len(array) == length
+
+
+def test_python_file_readinto_different_buffertype():
+    length = 5
+    data = b'some sample data longer than 10'
+    array = np.empty(length, dtype=np.uint16)
+    dst_buf = memoryview(array)
+    src_buf = BytesIO(data)
+    byte_length = array.nbytes
+
+    with pa.PythonFile(src_buf, mode='r') as f:
+        assert f.readinto(dst_buf) == byte_length
+
+        assert array[:byte_length].tobytes() == data[:byte_length]
+        assert len(array) == length
+
+
+def test_python_file_readline():
+    data = b'''line1
+    line2
+    line3
+    '''
+
+    buf = BytesIO(data)
+    buf2 = BytesIO(data)
+    with pa.PythonFile(buf, mode='r') as f:
+        for index in range(4):
+            assert f.readline() == buf2.readline()
+
+
+def test_python_file_readline_no_nl_at_eof():
+    data = b'''line1
+    line2
+    line3'''
+
+    buf = BytesIO(data)
+    buf2 = BytesIO(data)
+    with pa.PythonFile(buf, mode='r') as f:
+        for index in range(4):
+            assert f.readline() == buf2.readline()
+
+
+def test_python_file_readlines():
+    data = b'''line1
+    line2
+    line3
+    '''
+
+    buf = BytesIO(data)
+    buf2 = BytesIO(data)
+    with pa.PythonFile(buf, mode='r') as f:
+        assert f.readlines() == buf2.readlines()
+
+
+def test_python_file_readlines_with_hint():
+    data = b'''line1
+    line2
+    line3
+    '''
+
+    buf = BytesIO(data)
+    buf2 = BytesIO(data)
+    with pa.PythonFile(buf, mode='r') as f:
+        assert f.readlines(8) == buf2.readlines(8)
+
+
+def test_python_file_correct_abc():
+    with pa.PythonFile(BytesIO(b''), mode='r') as f:
+        assert isinstance(f, RawIOBase)
+        assert isinstance(f, IOBase)
+
+
+def test_python_file_correct_ducktyping():
+    with pa.PythonFile(BytesIO(b''), mode='r') as f:
+        assert hasattr(f, '__iter__')
```

Comment:
Instead of using `hasattr`, how about just iterating on the file?

Reply:
I wanted to test the failure mentioned in the JIRA ticket explicitly, i.e. have the exact same check as pandas does. The iterating over file is tested in `test_python_file_iterable`, too.

Reply:
I removed the test.